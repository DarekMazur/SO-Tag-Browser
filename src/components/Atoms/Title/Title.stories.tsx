import type { Meta, StoryObj } from '@storybook/react';
import { faker } from '@faker-js/faker';
import Title from './Title.tsx';

const meta: Meta<typeof Title> = {
  tags: ['autodocs'],
  component: Title,
  title: 'Atoms/Title',
};

export default meta;

type Story = StoryObj<typeof Title>;

export const Default: Story = {
  argTypes: {
    label: {
      control: 'text',
      description: 'Title content',
      name: 'label',
    },
  },
  args: {
    label: faker.lorem.sentence({ min: 2, max: 3 }),
  },
};
