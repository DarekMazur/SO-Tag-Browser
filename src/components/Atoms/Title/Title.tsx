import { Typography } from '@mui/material';

const Title = ({ label }: { label: string }) => {
  return (
    <Typography
      variant="h1"
      component="h1"
      padding="2rem"
      align="center"
      fontWeight="600"
      fontFamily="'Cinzel', serif"
    >
      {label}
    </Typography>
  );
};

export default Title;
