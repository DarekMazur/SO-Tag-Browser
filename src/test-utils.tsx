// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import { ReactNode } from 'react';
import AppProviders from './providers/AppProviders.tsx';

const AllTheProviders = ({ children }: { children: ReactNode }) => {
  return <AppProviders>{children}</AppProviders>;
};

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
const customRender = (ui, options?) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// eslint-disable-next-line import/no-extraneous-dependencies
export * from '@testing-library/react';

export { customRender as render };
