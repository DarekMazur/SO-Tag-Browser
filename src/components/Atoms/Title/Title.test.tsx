import { render, screen } from '../../../test-utils.tsx';
import Title from './Title.tsx';

describe('Tag link', () => {
  it('Renders the component', () => {
    const { container } = render(<Title label="Title label" />);
    expect(container).toMatchSnapshot();
  });
  it('Renders with prop', () => {
    render(<Title label="Title label" />);
    expect(screen.getByText(/Title/)).toBeInTheDocument();
  });
});
