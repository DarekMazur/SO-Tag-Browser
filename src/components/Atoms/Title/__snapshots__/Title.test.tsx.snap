// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tag link > Renders the component 1`] = `
<div>
  <h1
    class="MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter css-1aaknn6-MuiTypography-root"
  >
    Title label
  </h1>
</div>
`;
